### PR TITLE
Enable enhanced monitoring

### PR DESCRIPTION
**Pull request by robinbowes**
Creating the monitoring role in the module seems to work.

(README not updated)

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -110,3 +111,28 @@ resource "aws_security_group_rule" "allow_all_outbound" {
 
   security_group_id = "${aws_security_group.main_db_access.id}"
 }
+
+resource "aws_iam_role" "rds_monitoring" {
+  name               = "${var.monitoring_role_name}"
```
Comment (antonbabenko) [inline main.tf]: You can use `count = ${var.monitoring_interval == "0" ? 0 : 1}` on `aws_iam_role` and `aws_iam_role_policy_attachment` to not create resources if they are not needed

File: main.tf [hunk 2]
```diff
@@ -47,6 +47,7 @@ resource "aws_db_instance" "main_rds_instance" {
 
   # enhanced monitoring
   monitoring_interval = "${var.monitoring_interval}"
+  monitoring_role_arn = "${aws_iam_role.rds_monitoring.arn}"
```
Comment (antonbabenko) [inline main.tf]: `monitoring_role_arn = "${var.monitoring_interval == "0" ? "" : aws_iam_role.rds_monitoring.arn}"`

File: main.tf (hunk 1 above)
Comment (robinbowes) [reply main.tf]: Yes, I know, but I seem to recall that if you use the resource (as in the monitoring_role_arn) even if conditional, then it has to exist. I will check...

**Comment by robinbowes** [comment]
Yeah, with the current code, I get:

```
Error running plan: 1 error(s) occurred:

* module.mysql.aws_db_instance.main_rds_instance: 1 error(s) occurred:

* module.mysql.aws_db_instance.main_rds_instance: Resource 'aws_iam_role.rds_monitoring' not found for variable 'aws_iam_role.rds_monitoring.arn'
```

**Comment by robinbowes** [comment]
OK, I think this is ready to merge.

I've pruned it back to the most simple implementation possible - ie. just add the additional parameter.

The module user is responsible for making sure that the specified role exists.

**Comment by antonbabenko** [comment]
LGTM, sorry for the delay in response! Will you be able to adapt this PR to fit nicely in https://github.com/terraform-aws-modules/terraform-aws-rds and submit PR there instead?

**Comment by robinbowes** [comment]
https://github.com/terraform-aws-modules/terraform-aws-rds/pull/6

**Pull request closed without merging**